includes/Data/Patterns.php

```diff
@@ -252,4 +277,61 @@ public static function get_count_of_patterns() {
 		}
 		 return $theme_pattern_count;
 	}
+
+	public static function get_all_patterns_for_slug($slug_keyword)
+     {
+          $active_theme = (\wp_get_theme())->get('TextDomain');
+
+          $block_patterns_registry = \WP_Block_Patterns_Registry::get_instance();
+          $all_patterns = $block_patterns_registry->get_all_registered();
+
+          $filtered_patterns = array();
+          foreach($all_patterns as $pattern_info) {
+               if( strpos($pattern_info['slug'], $active_theme) === 0 // making sure the slug name starts with the theme name
+                   && strpos($pattern_info['slug'], $slug_keyword) !== FALSE ) {
+                    $filtered_patterns[] = $pattern_info;
+               }
+          }
+          return $filtered_patterns;
+     }
+
+	 public static function get_default_header()
+     {
+          $active_theme = (\wp_get_theme())->get('TextDomain');
+
+          $header_menu_page_pattern = array_keys(self::get_theme_step_patterns()[$active_theme]['header-menu']);
```

Comment:
As you are using the Zeroth index as the default header we can resort to that itself without having a seperate endpoint and a function, as we already get the list and can easily make that decision.

Reply:
yup ... made that change